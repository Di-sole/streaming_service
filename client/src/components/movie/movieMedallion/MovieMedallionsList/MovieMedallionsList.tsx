import styles from './MovieMedallionsList.module.scss';
import MovieMedallionItem from '../MovieMedallionItem/MovieMedallionItem';
import Link from 'next/link';
import TextSquareUI from '@/components/UI/squares/TextSquareUI/TextSquareUI';
import ImgSquareUI from '@/components/UI/squares/ImgSquareUI/ImgSquareUI';
import { useTranslation } from 'next-i18next';
import IMovie from '@/models/IMovie';
import { useRouter } from 'next/router';

interface MovieMedallionsListProps {
    className?: string;
    movie: IMovie;
}

const MovieMedallionsList = ({ className, movie }: MovieMedallionsListProps) => {
    const { t } = useTranslation('moviesPage');
    const { locale } = useRouter();
    const actors = movie.persons.filter((persons) => persons.enProfession.includes('actor'));
    return (
        <div className={[styles.content, className].join(' ').trim()} data-testid={'movieMedallionsList'}>
            <MovieMedallionItem text={t('filterPanel.rating')} disabled={true}>
                <TextSquareUI value={Number(movie.ratingKp.toFixed(1))} />
            </MovieMedallionItem>
            {actors.slice(0, 5).map((actor) => (
                <Link href={`/persons/${actor.id}`} key={actor.id}>
                    <MovieMedallionItem
                        text={
                            locale == 'ru'
                                ? actor.name
                                    ? actor.name
                                    : actor.enName
                                : actor.enName
                                ? actor.enName
                                : actor.name
                        }
                    >
                        <ImgSquareUI person={actor} />
                    </MovieMedallionItem>
                </Link>
            ))}
        </div>
    );
};

export default MovieMedallionsList;
