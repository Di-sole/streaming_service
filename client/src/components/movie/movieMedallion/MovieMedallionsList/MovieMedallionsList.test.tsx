import { render, screen } from '@testing-library/react';
import MovieMedallionsList from './MovieMedallionsList';
import { movieTest } from '../../../../testAsserts/testItems';
import { act } from 'react-dom/test-utils';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('MOVIEMEDALLIONSLIST SNAPSHOTS TESTS', () => {
    test('MovieMedallionsList should not be changed', () => {
        act(() => {
            render(<MovieMedallionsList movie={movieTest} />);
        });

        const div = screen.getByTestId('movieMedallionsList');
        expect(div).toMatchSnapshot();
    });
});
