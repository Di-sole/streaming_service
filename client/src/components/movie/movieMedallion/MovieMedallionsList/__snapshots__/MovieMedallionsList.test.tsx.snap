// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIEMEDALLIONSLIST SNAPSHOTS TESTS MovieMedallionsList should not be changed 1`] = `
<div
  class="content"
  data-testid="movieMedallionsList"
>
  <div
    class="container container_inactive"
    data-testid="medallionItem"
  >
    <div
      class="container__content"
    >
      <div
        class="container__contentBlock"
      >
        <div
          class="value  value_high"
          data-testid="textSquare"
        >
          8.8
        </div>
      </div>
    </div>
    <div
      class="container__caption"
    >
      filterPanel.rating
    </div>
  </div>
  <a
    href="/persons/71427"
  >
    <div
      class="container"
      data-testid="medallionItem"
    >
      <div
        class="container__content"
      >
        <div
          class="container__contentBlock"
        >
          <img
            alt="Франсуа Клюзе"
            class="image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_71427.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
      </div>
      <div
        class="container__caption"
      >
        François Cluzet
      </div>
    </div>
  </a>
  <a
    href="/persons/3084680"
  >
    <div
      class="container"
      data-testid="medallionItem"
    >
      <div
        class="container__content"
      >
        <div
          class="container__contentBlock"
        >
          <img
            alt="Абса Дьяту Тур"
            class="image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_3084680.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_3084680.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_3084680.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
      </div>
      <div
        class="container__caption"
      >
        Absa Diatou Toure
      </div>
    </div>
  </a>
  <a
    href="/persons/1960244"
  >
    <div
      class="container"
      data-testid="medallionItem"
    >
      <div
        class="container__content"
      >
        <div
          class="container__contentBlock"
        >
          <img
            alt="Мари-Лор Дикуру"
            class="image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1960244.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1960244.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1960244.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
      </div>
      <div
        class="container__caption"
      >
        Marie-Laure Descoureaux
      </div>
    </div>
  </a>
  <a
    href="/persons/2228474"
  >
    <div
      class="container"
      data-testid="medallionItem"
    >
      <div
        class="container__content"
      >
        <div
          class="container__contentBlock"
        >
          <img
            alt="Йен Фенелон"
            class="image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_2228474.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_2228474.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_2228474.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
      </div>
      <div
        class="container__caption"
      >
        Ian Fenelon
      </div>
    </div>
  </a>
  <a
    href="/persons/1277398"
  >
    <div
      class="container"
      data-testid="medallionItem"
    >
      <div
        class="container__content"
      >
        <div
          class="container__contentBlock"
        >
          <img
            alt="Бенжамин Барош"
            class="image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1277398.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1277398.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Factor_iphone%2Fiphone360_1277398.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
        </div>
      </div>
      <div
        class="container__caption"
      >
        Benjamin Baroche
      </div>
    </div>
  </a>
</div>
`;
